docs/source/conf.py

```diff
@@ -134,6 +134,7 @@
         "kedro.io.core.DatasetError",
         "kedro.io.core.Version",
         "kedro.io.data_catalog.DataCatalog",
+        "kedro.io.kedro_data_catalog.KedroDataCatalog",
```

Comment:
Do you mean moving `KedroDataCatalog ` class into `data_catalog` instead of creating a new file?

Reply:
If yes, I'd prefer to keep them separately. Otherwise, `kedro.io.data_catalog` becomes really long, and it's hard to compare two implementations. Since we import both from `kedro.io`, there shouldn't be any problems when removing one of them later.